content/en/tutorials/using-terraform-with-testcontainers-and-localstack/index.md

```diff
@@ -0,0 +1,351 @@
+
+- Non-Existent Product Handling: Validates the system's response to a request for a non-existent product, ensuring the Lambda function properly returns the appropriate error message "Product not found".
+
+![architecture-diagram](architecture-diagram-test.png)
+
+Since the app runs entirely inside the LocalStack container, an HTTP client is used to make calls against the service.
+
+```java
+  @Test
+    @Order(1)
+    void testSuccessfulPostAction() {
+
+        var postUrl =
+                localStackEndpoint + "/restapis/" + apiGWId + "/dev/_user_request_/productApi";
+
+        var expectedResponse =
+                "Product added/updated successfully.";
+
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+
+            // add headers to a POST request
+            var httpPost = new HttpPost(postUrl);
+            httpPost.setHeader(new BasicHeader("Content-Type", "application/json"));
+            // create the JSON request body
+            var jsonRequestBody = "{\n" +
+                    "  \"id\": \"34534\",\n" +
+                    "  \"name\": \"EcoFriendly Water Bottle\",\n" +
+                    "  \"description\": \"A durable, eco-friendly water bottle.\",\n" +
+                    "  \"price\": \"29.99\"\n" +
+                    "}";
+
+            // set the request body
+            var entity = new StringEntity(jsonRequestBody);
+            httpPost.setEntity(entity);
+            // execute the request
+            try (CloseableHttpResponse response = httpClient.execute(httpPost)) {
+                String responseBody = EntityUtils.toString(response.getEntity());
+                //assert 200 OK status & response message
+                Assertions.assertEquals(HttpStatus.SC_OK, response.getStatusLine().getStatusCode());
+                Assertions.assertEquals(expectedResponse, responseBody);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+```
+
+It is now incredibly straightforward to utilize our Terraform configuration file directly to construct the exact, production-ready environment needed for effective testing.
+
+### About owner permissions
+
+It's important to note that this extension is still new and primarily intended for straightforward Terraform configurations. It may suffer improvement changes in the future.
+If you mount a directory instead of a single file, the AWS Terraform provider will not be downloaded each time the `init` command runs, and any state files created will be in your host directory, 
+potentially requiring `sudo` to modify or delete. The reason for this is that the container user that creates these files is `root`, and on Linux systems this will propagate 
+to your local files. MacOS, on the other hand, will not allow this to happen and your locally created files will belong to your user. Here's how it looks like:
```

Comment:
there's repetition here with a previous paragraph where we already explained this.